app/src/main/java/com/futurice/freesound/feature/common/ui/adapter/IListItemComparator.java

```diff
@@ -0,0 +1,26 @@
+package com.futurice.freesound.feature.common.ui.adapter;
+
+import com.futurice.freesound.feature.common.DisplayableItem;
+
+public interface IListItemComparator {
+
+    /**
+     * Decides whether two {@link DisplayableItem} represent the same Item.
+     * For example, if your items have unique ids, this method should check their id equality.
+     *
+     * @return True if the two items represent the same object or false if they are different.
+     */
+    boolean areItemsTheSame(final DisplayableItem item1, final DisplayableItem item2);
+
+    /**
+     * Checks whether the visual representation of two {@link DisplayableItem}s is the same.
+     *
+     * This method is called only if {@link #areItemsTheSame(DisplayableItem, DisplayableItem)}
```

Comment:
I'm not sure you can usefully declaring how/when this method is called. I think you're better off just declaring it's purpose and leave any comments about *when* this method is called to points where this is called.